rviz_common/include/rviz_common/ros_topic_display.hpp

```diff
@@ -55,6 +55,10 @@ static const rmw_qos_profile_t display_default =
   5,
   RMW_QOS_POLICY_RELIABILITY_RELIABLE,
   RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT,
+  RMW_QOS_DEADLINE_DEFAULT,
```

Comment:
to further future-proof this code - would it be better to set it to `rmw_qos_profile_default`, then modify the fields that are different?

Reply:
This struct is currently initialized statically. It will need a slightly bigger change to do something like, but I am willing to make that change.

Reply:
I can't say for certain if this pattern is happy here - but doing something like this would keep the change localized to a few lines right here

```
#include <iostream>

// e.g. default qos
struct Foo {
  int num = 1;
};

inline Foo make_default_foo() {
 Foo foo;
 foo.num = 2;
 return foo; 
}

static const A global_foo = make_default_foo();

int main() {
  assert(global_foo.num == 2);  // passes
  return 0;
}
```

Reply:
Turns out this is valid C++, but not valid C. So the `rviz` package won't have any issues with this approach, but some other packages will.

Reply:
I don't see why that stops us from taking a more robust approach here.

C only accepts fully static initialization, yes. For those packages, we could accept this coupling (rcl is much more closely tied to rmw than rviz is, so it makes more sense to have to change it). However, even for C-based packages, one could change to just using the function instead of using a static object.

```
// e.g instead of
static const foo_t my_foo = {
  0
};

// do
static inline foo_t get_my_foo() {
  foo_t my_foo = default_foo;
  my_foo.baz = 10;
  return my_foo;
}
```